poutyne/framework/callbacks/tracker.py

```diff
@@ -0,0 +1,176 @@
+            layer_gradient = layer_params.grad
+
+            abs_value_layer_gradient = layer_gradient.abs()
+
+            batch_layer_means.append(abs_value_layer_gradient.mean().cpu().detach().numpy())
+            batch_layer_min.append(layer_gradient.min().cpu().detach().numpy())
+            batch_layer_max.append(layer_gradient.max().cpu().detach().numpy())
+
+        batch_layer_means = np.array(batch_layer_means)
+        previous_mean = self.running_mean
+
+        self.running_mean = previous_mean + (batch_layer_means - previous_mean) / self.counter
+
+        self.running_m2 = self.running_m2 + (batch_layer_means - previous_mean) * (batch_layer_means -
+                                                                                   self.running_mean)
+
+        self.running_variance = self.running_m2 / (self.counter - 1) if self.counter > 1 else self.running_variance
+
+        batch_layer_min = np.array(batch_layer_min)
+        batch_layer_max = np.array(batch_layer_max)
+
+        self.running_min = np.minimum(batch_layer_min, self.running_min)
+        self.running_max = np.maximum(batch_layer_max, self.running_max)
+
+        self.counter += 1
+
+    def get_stats(self, layer_names: List[str]) -> Dict:
+        """
+        Get the accumulated statistics of the layers.
+        Note: This will reset the gradient tracker statistics values.
+        Args:
+            layer_names (List[str]): The names of the layer to get statistics from.
+        Returns:
+            A dictionary where the keys are the layer names and the values are the statistics of the layer.
+            The statistics is also a dictionary where the keys are the logged statistics
+            (mean, mean +/- std deviation, min and max) and the values are the corresponding statistic values.
+        """
+        formatted_stats = {}
+        for index, layer_name in enumerate(layer_names):
+            stats = {
+                "mean": self.running_mean[index],
+                "mean_std_dev_up": self.running_mean[index] + np.sqrt(self.running_variance[index]),
+                "mean_std_dev_down": self.running_mean[index] - np.sqrt(self.running_variance[index]),
+                "min": self.running_min[index],
+                "max": self.running_max[index]
+            }
+
+            formatted_stats.update({layer_name: stats})
+
+        self.reset()
+        return formatted_stats
+
+    def reset(self) -> None:
+        """
+        Reset the running mean, variance, min, max and counter values.
+        """
+        self.running_mean = np.zeros([self.number_layers], dtype="float32")
+        self.running_variance = np.zeros([self.number_layers], dtype="float32")
+        self.running_m2 = np.zeros([self.number_layers], dtype="float32")
+        self.running_min = np.zeros([self.number_layers], dtype="float32")
+        self.running_max = np.zeros([self.number_layers], dtype="float32")
+        self.counter = 1
```

Comment:
I would call this only `count` instead of `counter`.

Reply:
7be33b3